### ISSUE TITLE
Questionnaire crashes on choice options greater than 10 which opens QuestionnaireItemDialogSelectViewHolderFactory

### ISSUE THREAD

**Issue by maimoonak** [google/android-fhir#1453]
**Describe the bug**
Application crashes for questions having choices >= 10,

**Component**
SDC library

**To Reproduce**
1. Open a questionnaire with choices count > 10 which triggers the dialogue widget, and which in turn crashes. An example is https://pastebin.pl/view/cae6a75d, where clicking 'yes' on danger signs question makes questionnaire crash (use yes in all options to enable question)

**Expected behavior**
Should not crash and work on dropdown dialogue as well

`java.lang.UnsupportedOperationException: SavedStateViewModelFactory constructed with empty constructor supports only calls to create(modelClass: Class<T>, extras: CreationExtras).
        at androidx.lifecycle.SavedStateViewModelFactory.create(SavedStateViewModelFactory.kt:146)
        at androidx.lifecycle.SavedStateViewModelFactory.create(SavedStateViewModelFactory.kt:186)
        at dagger.hilt.android.internal.lifecycle.HiltViewModelFactory.create(HiltViewModelFactory.java:111)
        at androidx.lifecycle.ViewModelProvider$Factory.create(ViewModelProvider.kt:81)
        at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.kt:182)
        at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.kt:151)
        at androidx.lifecycle.ViewModelLazy.getValue(ViewModelLazy.kt:53)
        at androidx.lifecycle.ViewModelLazy.getValue(ViewModelLazy.kt:35)
        at com.google.android.fhir.datacapture.views.QuestionnaireItemDialogSelectViewHolderFactory$getQuestionnaireItemViewHolderDelegate$1.bind$lambda-1(QuestionnaireItemDialogSelectViewHolderFactory.kt:61)
        at com.google.android.fhir.datacapture.views.QuestionnaireItemDialogSelectViewHolderFactory$getQuestionnaireItemViewHolderDelegate$1.access$bind$lambda-1(QuestionnaireItemDialogSelectViewHolderFactory.kt:48)
        at com.google.android.fhir.datacapture.views.QuestionnaireItemDialogSelectViewHolderFactory$getQuestionnaireItemViewHolderDelegate$1$bind$1.invokeSuspend(QuestionnaireItemDialogSelectViewHolderFactory.kt:70)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.internal.DispatchedContinuationKt.resumeCancellableWith(DispatchedContinuation.kt:367)
        at kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable(Cancellable.kt:30)
        at kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable$default(Cancellable.kt:25)
        at kotlinx.coroutines.CoroutineStart.invoke(CoroutineStart.kt:110)
        at kotlinx.coroutines.AbstractCoroutine.start(AbstractCoroutine.kt:126)
        at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch(Builders.common.kt:56)
        at kotlinx.coroutines.BuildersKt.launch(Unknown Source:1)
        at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch$default(Builders.common.kt:47)
        at kotlinx.coroutines.BuildersKt.launch$default(Unknown Source:1)
        at com.google.android.fhir.datacapture.views.QuestionnaireItemDialogSelectViewHolderFactory$getQuestionnaireItemViewHolderDelegate$1.bind(QuestionnaireItemDialogSelectViewHolderFactory.kt:68)
        at com.google.android.fhir.datacapture.views.QuestionnaireItemViewHolder.bind(QuestionnaireItemViewHolderFactory.kt:63)
        at com.google.android.fhir.datacapture.QuestionnaireItemAdapter.onBindViewHolder(QuestionnaireItemAdapter.kt:99)
        at com.google.android.fhir.datacapture.QuestionnaireItemAdapter.onBindViewHolder(QuestionnaireItemAdapter.kt:43)
        at androidx.recyclerview.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:7254)
        at androidx.recyclerview.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:7337)
        at androidx.recyclerview.widget.RecyclerView$Recycler.tryBindViewHolderByDeadline(RecyclerView.java:6194)
        at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6460)
        at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6300)
        at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6296)
        at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2330)
        at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1631)
        at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1591)
        at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:668)
        at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4309)
`

https://github.com/google/android-fhir/blob/master/datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireItemAdapter.kt#L151 making app crash

**Comment by PallaviGanorkar** [comment]
@maimoonak I am not able to reproduce this issue, Please find attached Video for details
[Logs_1453.webm](https://user-images.githubusercontent.com/13957417/178259942-11ec58eb-d566-4e85-b674-b1b8c7c5a67f.webm)


Could you please share exact steps to reproduce this issue along with build version, device details and the video if possible.

**Comment by aurangzaibumer** [comment]
> @maimoonak I am not able to reproduce this issue, Please find attached Video for details [Logs_1453.webm](https://user-images.githubusercontent.com/13957417/178259942-11ec58eb-d566-4e85-b674-b1b8c7c5a67f.webm)
> 
> Could you please share exact steps to reproduce this issue along with build version, device details and the video if possible.

This issue was reproducing on **FHIR CORE (Implementer Application)** application but not on **FHIR SDK**. Me & @shoaibmushtaq25 are currently investigating the issue.

**Comment by aurangzaibumer** [comment]
> > @maimoonak I am not able to reproduce this issue, Please find attached Video for details [Logs_1453.webm](https://user-images.githubusercontent.com/13957417/178259942-11ec58eb-d566-4e85-b674-b1b8c7c5a67f.webm)
> > Could you please share exact steps to reproduce this issue along with build version, device details and the video if possible.
> 
> This issue was reproducing on **FHIR CORE (Implementer Application)** application but not on **FHIR SDK**. Me & @shoaibmushtaq25 are currently investigating the issue.

will share our findings in a while. thanks

**Comment by shoaibmushtaq25** [comment]
@PallaviGanorkar 
Previously, we could reproduce the issue on the FHIR core. As I and @aurangzaibumer did more investigation, this issue is not occurring in FHIR Core(Implementer App) after recent changes in FHIR-core and FHIR-SDK.  We tested this issue with the [given questionnaire](https://pastebin.pl/view/cae6a75d) shared by @maimoonak above in the description of this issue.
So it looks like this issue doesn't exist anymore.
We tried different scenarios to test it including Single-select drop-down and Multi-select drop-down with more than 10 choice answer items.
For more details, the Videos are attached here.

1. [Single-select drop-down video](https://user-images.githubusercontent.com/12672919/178801701-3bad9612-c777-47af-b9df-e1d159d3a491.mp4)
2. [Multi-select drop-down video](https://user-images.githubusercontent.com/12672919/178801941-33f0b201-db6d-4955-b737-1b75e769ab19.mp4)

cc: @aurangzaibumer @f-odhiambo @maimoonak @jingtang10 @Tarun-Bhardwaj
